docs/devel/faster_reviews.md

```diff
@@ -154,7 +163,18 @@ commit and re-push.  Your reviewer can then look at that commit on its own - so
 much faster to review than starting over.
 
 We might still ask you to clean up your commits at the very end, for the sake
-of a more readable history.
+of a more readable history, but don't do this until asked, typically at the point
+where the PR would otherwise be tagged LGTM.
+
+General squashing guidelines:
```

Comment:
A commit, as much as possible, should be a single logical change.  Commits should always have a good title line (<70 characters) and include an additional description paragraph describing in more detail the change intended.